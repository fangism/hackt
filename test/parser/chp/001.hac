// "001.hac" should fail, channel should not contain channel

defproc empty(chan?(chan(bool)) L; chan!(bool) R);

