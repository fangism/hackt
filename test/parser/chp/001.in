// "001.in" should fail, channel should not contain channel

defproc empty(chan?(chan(bool)) L; chan!(bool) R);

