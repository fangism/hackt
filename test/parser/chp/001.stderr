At: "001.hac":3:
parse error: syntax error
parser stacks:
state	value
#STATE#	(null) 
#STATE#	(null) 
#STATE#	(null) 
#STATE#	keyword: defproc [3:1..7]
#STATE#	identifier: empty [3:9..13]
#STATE#	( [3:14]
#STATE#	(chan-type) [3:15..19]
#STATE#	( [3:20]
#STATE#	keyword: chan [3:21..24]
in state #STATE#, possible rules are:
	data_type_ref_list_optional_in_parens : '(' . data_type_ref_list_optional ')'  (#RULE#)
	data_type_ref_list_optional : .  (#RULE#)
acceptable tokens are: 
	')' (reduce)
	ID (shift)
	SCOPE (shift)
	INT_TYPE (shift)
	BOOL_TYPE (shift)
